src/test/java/io/github/polysantiago/spring/rest/RestClientHateoasTest.java

```diff
@@ -75,6 +78,15 @@ public Object postProcessAfterInitialization(Object bean, String beanName) throw
             };
         }
 
+        @Bean
```

Comment:
This is actually not needed. Since Spring Hateoas 0.25.0, there's no custom ObjectMapper and it uses the one created per default. The problem is that it registers it for its own converter (`TypeConstrainedMappingJackson2HttpMessageConverter`) which does not read `Optional` and passes it on to the context's `MappingJackson2HttpMessageConverter` which is registered. However, this converter doesn't have the `Jackson2HalModule` registered. To fix this test, we can basically let Spring Boot register the module for the default ObjectMapper.
```
  @Bean
  Module halModule() {
    return new Jackson2HalModule();
   }
```

Reply:
Btw, the `BeanPostProcessor` can also be removed from the test.